provider/datagen/src/source.rs

```diff
@@ -211,6 +231,30 @@ impl SourceData {
     }
 }
 
+fn follow_redirect_and_get_last_element(redirector: &str) -> Result<String, DataError> {
+    let response = reqwest::blocking::Client::new()
+        .head(redirector)
+        .send()
+        .map_err(|e| DataError::custom("reqwest error").with_display_context(&e))?;
+    Ok(response
+        .url()
+        .path()
+        .split('/')
+        .next_back()
+        .expect("split is non-empty")
+        .to_string())
+}
+
+fn download(resource: &str) -> Result<PathBuf, DataError> {
+    cached_path::CacheBuilder::new()
+        .freshness_lifetime(u64::MAX)
+        .progress_bar(None)
```

Comment:
Question: why no progress bar?

Reply:
I hope we're downloading only once? I feel like we should block datagen while downloads are in progress?

Reply:
Yeah obviously. I mean the threads that need the data that's being downloaded are blocked on the `RwLock`, but threads that already have their data can continue.

Reply:
What you suggest would be quite complicated to do, because then we'd have to keep track of which markers require which data. Segmenter for example will never fail due to missing data because the data is bundled, so that would just start running while others will fail. Then we'd have to restart the failed ones somehow, but then if there's a problem with downloading, then we properly fail?

Effectively this will put the downloads in the first couple of lines, maybe some property keys are already being generated, but I don't think this a big deal. If your internet is slow enough the downloads will be at the beginning, with a message that they're starting and a message that they're done.

Reply:
ok.

I'm also not opposed to going back to eagerly downloading both CLDR and ICUExportData. If you care about not downloading more than you need, then set your own CLI arguments.

We can continue discussing this later. No need to block the PR

Reply:
The lazy download is the reason I made this PR. It's required for `--keys none` to not hit the network.

Reply:
ok. Let's land it like this and revisit after people have gotten more of a chance to experience the new behavior